pkg/block/local/adapter.go

```diff
@@ -79,18 +81,33 @@ func resolveNamespace(obj block.ObjectPointer) (block.QualifiedKey, error) {
 	return qualifiedKey, nil
 }
 
+// verifyPath ensures that p is under the directory controlled by this adapter.  It does not
+// examine the filesystem and can mistakenly error out when symbolic links are involved.
+func (l *Adapter) verifyPath(p string) error {
+	if !strings.HasPrefix(filepath.Clean(p), l.path) {
```

Comment:
Isn't better to disallow any path traversal attempts, like
```suggestion
	if filepath.Clean(p) != p {
```

Reply:
I don't think so.  E.g. the prefix itself might not be clean, or tierFS might be able to use it, etc.

Also doing this means that all callers must ensure they stitch paths onto the prefix.  I'd rather not trust callers in this way.